### PR TITLE
Hardware section and node diagram

### PR DESCRIPTION
**Pull request by sfarr-epcc**
First draft of ARCHER2 hardware section. Contains an architecture diagram from previous training courses and a new node diagram.

### CODE REVIEW

**Review by kevinstratford — COMMENTED** [review]
Couple of points:

Probably wouldn't say, under SMT, "each core has two threads", threads being a software entity. Prefer something like "each core can accommodate two threads in hardware". System software (slurm) often reports "256 CPUs". 

Where does the 204 GB/s memory bandwidth figure come from?

**Comment by sfarr-epcc** [comment]
Most of the numbers come from the slides I saved from this talk: 
https://www.archer2.ac.uk/training/courses/210420-efficient-use/

File: docs/user-guide/hardware.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+- Nodes are organized into groups.
+    - 128 Nodes in a group.
+    - Electrical links between Network Interface Card (NIC) and switch.
+    - 16 switches per group.
+    - 2x NIC per node.
+    - All-to-all connection amongst switches in a group using electrical links.
+
+- All-to-all connection between groups using optical links.
+    - 2 groups per ARCHER2 Cabinet.
```
Comment (aturner-epcc) [inline docs/user-guide/hardware.md:51]: @sfarr-epcc I just wanted to check that this description has been run past HPE to check they are happy with it. If not, can you just run a quick check past Harvey and Martin?

**Review by aturner-epcc — COMMENTED** [review]
I think we should also add a link to the Next Platform article mentioned in #59 as a place to look for further information. Maybe also a link to the wikichip page for the ARCHER2 processors (https://en.wikichip.org/wiki/amd/epyc/7742). Are you able to do this?

**Review by aturner-epcc — APPROVED** [review]
Looks fine now this has been checked with HPE

**Pull request merged**